src/core/Akka/Configuration/Hocon/HoconObject.cs

```diff
@@ -177,6 +177,33 @@ private string QuoteIfNeeded(string text)
             }
             return text;
         }
+
+        public void Merge(HoconObject other)
+        {
+            var thisItems = Items;
+            var otherItems = other.Items;
+
+            foreach (var otherItem in otherItems)
+            {
+                if (thisItems.ContainsKey(otherItem.Key))
+                {
+                    //other key was present in this object.
+                    //if we have a value, just ignore the other value, unless it is an object
+                    var thisItem = thisItems[otherItem.Key];
+
+                    //if both values are objects, merge them
+                    if (thisItem.IsObject() && otherItem.Value.IsObject())
+                    {
+                        thisItem.GetObject().Merge(otherItem.Value.GetObject());
+                    }
+                }
+                else
+                {
+                    //other key was not present in this object, just copy it over
+                    Items.Add(otherItem.Key,otherItem.Value);
+                }
+            }            
+        }
```

Comment:
This makes it possible to merge Hocon Objects. which could eventually be used to rewrite the `WithFallback` support.
Right now it is used for the Hocon Include support, merging the sub config with the object that does the include.